cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/IntraBrokerDiskUsageDistributionGoal.java

```diff
@@ -479,32 +455,22 @@ private void rebalanceBySwappingLoadIn(Disk disk,
 
     while (!candidateDiskPQ.isEmpty()) {
       Disk candidateDisk = candidateDiskPQ.poll();
-      candidateDisk.trackSortedReplicas(name(),
-                                        ReplicaSortFunctionFactory.selectOnlineReplicas(),
-                                        ReplicaSortFunctionFactory.sortByMetricGroupValue(RESOURCE.name()));
-      for (Iterator<Replica> iterator = disk.trackedSortedReplicas(name()).sortedReplicas().iterator();
+      for (Iterator<Replica> iterator = disk.trackedSortedReplicas(replicaSortName(this, false, false)).sortedReplicas(false).iterator();
```

Comment:
Can we add a unit test to ensure that any unintended future change to this behavior can be captured via tests rather than potentially during run time?